client/lib/url/url-type.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+import { URL as URLString } from 'types';
+import { Falsey } from 'utility-types';
+
+export enum URL_TYPE {
+	// A complete URL, with (at least) protocol and host.
+	// E.g. `http://example.com` or `http://example.com/path`
+	ABSOLUTE = 'ABSOLUTE',
+	// A URL with no protocol, but with a host.
+	// E.g. `//example.com` or `//example.com/path`
+	PROTOCOL_RELATIVE = 'PROTOCOL_RELATIVE',
```

Comment:
_Technically_, I think this should be `SCHEME_RELATIVE` 🤓 

https://url.spec.whatwg.org/#scheme-relative-special-url-string

Reply:
Oooh, good job finding this spec! I'll switch to the proper naming used there  🤓

Reply:
Oooh, good job finding this spec! I'll switch to the proper naming used there for everything  🤓